user/views.py

```diff
@@ -0,0 +1,60 @@
+import json
+import re
+import bcrypt
+
+from django.http                       import JsonResponse, HttpResponse
+from django.views                      import View
+
+from user.models                       import (
+    User
+)
+
+email_regex              = re.compile(r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+password_regex           = re.compile(r'^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$')
+MINIMUM_PASSWORD_LENGTH  = 8
+
+class SignUpView(View):
+
+    def post(self,request):
+        try:
+            data         = json.loads(request.body)
+
+            email        = data['email']
+            fullname     = data['fullname']
+            password     = data['password']
+            maker_info   = data.get('maker_info', None)
+
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({"message" : "EMAIL_EXISTS"}, status=400)
+
+            if not email:
+                return JsonResponse({"message" : "EMAIL_REQUIRED"}, status=400)
+
+            if not fullname:
+                return JsonResponse({"message" : "FULLNAME_REQUIRED"}, status=400)
+
+            if not password:
+                return JsonResponse({"message" : "PASSWORD_REQUIRED"}, status=400)
+
+            if len(password) < MINIMUM_PASSWORD_LENGTH:
+                return JsonResponse({"message" : "PASSWORD_MUST_BE_LONGER_THAN_EIGHT_LETTERS"}, status=400)
+
+            if not email_regex.search(email):
+                return JsonResponse({"message" : "EMAIL_VALIDATION"}, status=400)
```

Comment:
`email_regex.search(email)`의 결과값이 Falsy value 인 경우 유효하지 않은 이메일 주소라는 뜻이니, JSON response 의 메세지를 조금 더 명확하게 지어주시는게 좋겠네요. 아래처럼요!
```suggestion
            if not email_regex.search(email):
                return JsonResponse({"message" : "INVALID_EMAIL"}, status=400)
```

Reply:
헉 지금 수정하겠습니다! 명확한... 메세지... 메모... 📝